compiler/noirc_frontend/src/lexer/lexer.rs

```diff
@@ -21,6 +21,20 @@ pub struct Lexer<'a> {
 
 pub type SpannedTokenResult = Result<SpannedToken, LexerErrorKind>;
 
+pub(crate) fn from_spanned_token_result(
+    x: &SpannedTokenResult,
+) -> Result<(usize, &Token, usize), LexerErrorKind> {
+    x.as_ref()
```

Comment:
I think we can remove this `as_ref()`, you can then return a `Result<(usize, Token, usize), LexerErrorKind>` and you can avoid cloning in the map_err

Reply:
I'm using the lifetime of the `Vec` for the `Tok`'s lifetime, when it'd normally be the input `str`'s lifetime, passed through the lexing process.

Just removing `as_ref`, the compiler suggests making `x` a reference so that the lifetime gets passed through to `Tok`:
```bash
help: consider adjusting the signature so it borrows its argument
    -->  compiler/noirc_frontend/src/lexer/lexer.rs:25:19
     |
25   |     token_result: &SpannedTokenResult,
     |                   +

```
(can paste the rest of the error if you'd like)